In [54]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, adjusted_rand_score
pd.set_option('display.max_colum', None)
pd.set_option('display.max_rows', 100)

In [2]:
advert = pd.read_csv('https://raw.githubusercontent.com/justmarkham/scikit-learn-videos/master/data/Advertising.csv',
                     usecols=['TV', 'Radio', 'Newspaper', 'Sales'])


In [3]:
advert.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3    9.3
3  151.5   41.3       58.5   18.5
4  180.8   10.8       58.4   12.9

In [4]:
x=advert[['TV','Radio','Newspaper']]

In [5]:
y=advert['Sales']

In [6]:
# initlize the model
model=LinearRegression()
model.fit(x,y)

LinearRegression()

In [34]:
print("Alpha value is:{}".format(model.intercept_))
print("betas value is:{}".format(model.coef_))
print("Betas Value according to Independent variable")
print("The value of TV is:{}".format(model.coef_[0]*100))
print("__________________")
print("The value of Radio is:{}".format(model.coef_[1]*100))
print("__________________")

print("The value of Newspaper is:{}".format(model.coef_[2]*100))

Alpha value is:2.9388893694594067
betas value is:[ 0.04576465  0.18853002 -0.00103749]
Betas Value according to Independent variable
The value of TV is:4.576464545539761
__________________
The value of Radio is:18.853001691820463
__________________
The value of Newspaper is:-0.10374930424762972


**Obervation**
- In this value we clear cut that if we invest more money in **Radio** advertisement then we genearte more revenue
- so our linear equation is **Sales**=18.85* Radio+4.57* TV -0.10* Newspaper + 2.93

In [38]:
pd.DataFrame(model.predict(x))

0
0    20.523974
1    12.337855
2    12.307671
3    17.597830
4    13.188672
..         ...
195   5.370342
196   8.165312
197  12.785921
198  23.767321
199  15.173196

[200 rows x 1 columns]

In [41]:
newX=[[100,300,20]]

In [42]:
model.predict(newX)

C:\Users\Sanwal Khan\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([64.05360913])

In [43]:
newX2=[[0,0,0]]
model.predict(newX2)

C:\Users\Sanwal Khan\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([2.93888937])

0      22.1
1      10.4
2       9.3
3      18.5
4      12.9
       ... 
195     7.6
196     9.7
197    12.8
198    25.5
199    13.4
Name: Sales, Length: 200, dtype: float64

In [48]:
y_pred=model.predict(x)

In [49]:
r2=r2_score(y,y_pred)

In [53]:
print('r2:',r2*100)

r2: 89.72106381789521


In [55]:
adj=adjusted_rand_score(y,y_pred)

C:\Users\Sanwal Khan\anaconda3\lib\site-packages\sklearn\metrics\cluster\_supervised.py:65: UserWarning: Clustering metrics expects discrete values but received continuous values for label, and continuous values for target
  warnings.warn(msg, UserWarning)


In [56]:
adj

0.0